Zilon.Core/Zilon.Core/PersonModules/CombatStatsModule.cs

```diff
@@ -25,5 +34,267 @@ public CombatStatsModule()
+                    var stat = _evolutionModule.Stats.SingleOrDefault(x => x.Stat == bonusItem.Key);
+                    if (stat != null)
+                    {
+                        ApplyBonusToStat(bonusItem.Value, stat);
+                    }
+                }
+
+                foreach (var statItem in _evolutionModule.Stats)
+                {
+                    statItem.Value = (float)Math.Round(statItem.Value, 1);
+                }
+            }
+
+            RecalculatePersonArmor();
+        }
+
+        /// <summary>
+        /// Применение бонуса к характеристике навыка.
+        /// </summary>
+        /// <param name="bonusValue"> Величина бонуса. </param>
+        /// <param name="stat"> Характеристика навыка. </param>
+        private static void ApplyBonusToStat(float bonusValue, SkillStatItem stat)
+        {
+            stat.Value += stat.Value * bonusValue;
+
+            if (stat.Value <= 1)
+            {
+                stat.Value = 1;
+            }
+        }
+
+        /// <summary>
+        /// Расчёт бонусов, которые дают перки.
+        /// </summary>
+        /// <param name="bonusDict"> Текущее состояние бонусов. </param>
+        private void CalcPerkBonuses(Dictionary<SkillStatType, float> bonusDict)
+        {
+            var archievedPerks = _evolutionModule.GetArchievedPerks();
+            foreach (var archievedPerk in archievedPerks)
+            {
+                var currentLevel = archievedPerk.CurrentLevel;
+                var currentLevelScheme = archievedPerk.Scheme.Levels[currentLevel.Primary];
+
+                if (currentLevelScheme.Rules == null)
+                {
+                    continue;
+                }
+
+                for (var i = 0; i <= currentLevel.Sub; i++)
+                {
+                    foreach (var rule in currentLevelScheme.Rules)
+                    {
+                        CalcRuleBonuses(rule, bonusDict);
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Расчёт бонуса, который даёт правило перка.
+        /// </summary>
+        /// <param name="rule"> Правило перка. </param>
+        /// <param name="bonusDict"> Текущее состояние бонусов. </param>
+        private static void CalcRuleBonuses(PerkRuleSubScheme rule, Dictionary<SkillStatType, float> bonusDict)
+        {
+            switch (rule.Type)
+            {
+                case PersonRuleType.Melee:
+                    AddStatToDict(bonusDict, SkillStatType.Melee, rule.Level, PersonRuleDirection.Positive);
+                    break;
+
+                case PersonRuleType.Ballistic:
+                    AddStatToDict(bonusDict, SkillStatType.Ballistic, rule.Level, PersonRuleDirection.Positive);
+                    break;
+
+                    //case PersonRuleType.Undefined:
+                    //default:
+                    //    throw new ArgumentOutOfRangeException($"Тип правила перка {rule.Type} не поддерживается.");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/manual/kreghek/Zilon_Roguelike/pullRequest?prid=5480188)